src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.cs

```diff
@@ -302,12 +274,10 @@ protected CSharpParseOptions CreateParseOptions()
             var languageVersion = CompilationOptionsConversion.GetLanguageVersion(GetStringOption(CompilerOptions.OPTID_COMPATIBILITY, defaultValue: ""))
                                   ?? CSharpParseOptions.Default.LanguageVersion;
 
-            return new CSharpParseOptions(
-                kind: SourceCodeKind.Regular,
-                languageVersion: languageVersion,
-                preprocessorSymbols: symbols.AsImmutable(),
-                documentationMode: documentationMode)
-                .WithFeatures(GetParsedCommandLineArguments().ParseOptions.Features);
+            return options.WithKind(SourceCodeKind.Regular)
```

Comment:
We generally avoided doing all the .With calls to avoid extra garbage creation, if I recall.